File: 1. Two Sum/Solution.cs
 public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        IDictionary<int, int> seen = new Dictionary<int, int>();

        for (int i = 0; i < nums.Length; i++) {
            int num = nums[i];
            int diff = target - num;
            if (seen.TryGetValue(diff, out int index)) {
                return new int[]{seen[diff], i};
            }
            seen[num] = i;
        }
        return new int[]{};
    }
}

File: 1. Two Sum/solution.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  const seen = {};

  for (let i = 0; i < nums.length; i++) {
    const num = nums[i];
    const diff = target - num;
    if (seen[diff] !== undefined) {
      return [seen[diff], i];
    }
    seen[num] = i;
  }
  return [];
};

// Time complexity: O(n)
// Space complexity: O(n)
